patches/CreatePBP.cmake

```diff
@@ -0,0 +1,66 @@
+# File defining macro outputting PSP-specific EBOOT.PBP out of passed executable target.
+# Copyright 2020 - Daniel 'dbeef' Zalega
+#
+# Args:
+# TARGET - defined by an add_executable call before calling create_pbp_file
+# ICON_PATH - optional, absolute path to .png file, 144x82
+# BACKGROUND_PATH - optional, absolute path to .png file, 480x272
+# PREVIEW_PATH - optional, absolute path to .png file, 480x272
+#
+cmake_minimum_required(VERSION 3.10)
+
+macro(create_pbp_file)
+
+    set(oneValueArgs TARGET ICON_PATH BACKGROUND_PATH PREVIEW_PATH)
+    cmake_parse_arguments("ARG" "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    # As pack-pbp takes undefined arguments in form of "NULL" string,
+    # set each undefined macro variable to such value:
+    foreach(arg ${oneValueArgs})
+        if (NOT DEFINED ARG_${arg})
+            set(ARG_${arg} "NULL")
+        endif()
+    endforeach()
+
+    add_custom_command(TARGET ${ARG_TARGET} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E make_directory
+            $<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact
+            COMMENT "Creating psp_artifact directory."
+    )
+
+    add_custom_command(TARGET ${ARG_TARGET} POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different
+            $<TARGET_FILE:${ARG_TARGET}>
+            $<TARGET_FILE_DIR:${ARG_TARGET}>/psp_artifact
+            COMMENT "Copying ELF to psp_arfitact directory."
+    )
+
```

Comment:
Could use absolute paths for the tools here. This way they do not need to be available in the `$PATH`.
You already define them in the toolchain file, e.g. `PACK_PBP`, `FIXUP`.

Reply:
Good idea, done.